e2e/libs/utility/utility.py

```diff
@@ -1,31 +1,184 @@
-        config.load_kube_config()  # for local environment
-        # config.load_incluster_config()  # for in-cluster environment
+        contexts, active_context = config.list_kube_config_contexts(globalVars.variables["KUBECONFIG_PATH"])
+        if not contexts:
+            raise Exception("cannot find any context in kube-config file.")
+
+        contexts = [context['name'] for context in contexts]
+        active_index = contexts.index(active_context['name'])
+        option, _ = pick(contexts, title="Pick the context to load",
+                        default_index=active_index)
+
+        kubecfg = config.load_kube_config(context=option)  # for local environment
+        globalVars.K8S_API_CLIENT = client.CoreV1Api(api_client = kubecfg)
+        globalVars.K8S_CR_API_CLIENT = client.CustomObjectsApi(api_client = kubecfg)
+        globalVars.K8S_APP_API_CLIENT = client.AppsV1Api(api_client = kubecfg)
+
+        logging.debug("Active host is %s" % configuration.Configuration().host)
+
+    @classmethod
+    def get_k8s_core_api_client(cls):
+        logging.info("getting K8S core API client")
+        cls().init_k8s_api_client()
+        return client.CoreV1Api()
 
+    @classmethod
     def get_longhorn_client(cls):
-        print('get_longhorn_client')
+        logging.info("getting Longhorn client")
         # manually expose longhorn client node port
         # otherwise the test is needed to be run in in-cluster environment
         # to access longhorn manager cluster ip
-        longhorn_client_url = os.getenv('LONGHORN_CLIENT_URL')
+        longhorn_client_url = globalVars.variables["LONGHORN_CLIENT_URL"]
         longhorn_client = from_env(url=f"{longhorn_client_url}/v1/schemas")
-        print(longhorn_client)
         return longhorn_client
+
+    @classmethod
+    def ssh_and_exec_cmd(cls, host, cmd):
+        if host == "" or cmd == "":
+            raise Exception(f'empty parameters: host={host}, cmd={cmd}')
+
+        logging.info(f"ssh into {host} and execute commands {cmd}")
+
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        cls().ssh_connect_with_retry(ssh, host, 0)
+        chan = ssh.get_transport().open_session()
+        chan.exec_command(cmd)
+
+        isExecCommand = True
+        while isExecCommand:
+            if chan.recv_ready():
+                logging.debug(f'response of command: ', chan.recv(4096).decode('ascii'))
+
+            if chan.exit_status_ready():
+                if chan.recv_stderr_ready() and chan.recv_exit_status() != 0:
+                    raise Exception('fail to execute command:', chan.recv_stderr(4096).decode('ascii'))
+
+                isExecCommand = False
+                logging.debug(f'success of command execution: {chan.recv_exit_status()}')
+                ssh.close()
+
+    def ssh_connect_with_retry(cls, ssh, ip_address, retries):
+        if retries > 3:
```

Comment:
Suggest to use for loop fo the retry instead of making this a self-recursive function.